homeassistant/components/light/tradfri.py

```diff
@@ -88,13 +95,19 @@ def turn_on(self, **kwargs):
         """Instruct the light to turn on."""
         if ATTR_BRIGHTNESS in kwargs:
             self._light.light_control.set_dimmer(kwargs.get(ATTR_BRIGHTNESS))
+        elif ATTR_RGB_COLOR:
+            # Handle Hue lights paired with the gateway
+            if self._light_data.hex_color is not None:
+                self._rgb_color = rgb_hex_to_rgb_list(
```

Comment:
Shouldn't you set the rgb passed in kwargs to the device? Currently you update hass rgb from the device. I assume you've added a set method in the tradfri library for rgb?